controllers/feeds/deleteAnonymousPostV2.js

```diff
@@ -0,0 +1,46 @@
+const getstreamService = require('../../services/getstream');
+const Getstream = require('../../vendor/getstream');
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ * @returns
+ */
+module.exports = async (req, res) => {
+  const {postId} = req.params;
+  const token = req.token;
+  const userId = req.userId;
+  const getstreamQueryResult = await getstreamService.getFeeds(token, 'user_anon', {
+    ids: [postId]
+  });
+  const [feed] = getstreamQueryResult?.results || {};
+  console.log('userId', feed?.actor?.id, userId);
+  if (feed?.actor?.id !== userId) {
+    return res.status(200).json({
+      success: false,
+      message: 'You are not authorized to delete this post'
+    });
+  }
+
+  try {
+    const getstreamDeleteResult = await getstreamService.deleteFeedById({
+      feedGroup: 'user_anon',
+      activityId: postId,
+      userId: userId,
+      token
+    });
+    await Getstream.feed.deleteNotificationFeed(userId, postId);
+    console.log('getstreamDeleteNotificationFeed');
+    // console.log(getstreamDeleteNotificationFeed);
+    return res.status(200).json({
+      success: getstreamDeleteResult ? true : false
+    });
+  } catch (e) {
+    console.log('error deleting anonymous post', e);
+    return res.status(200).json({
+      success: false,
+      message: e
+    });
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code seems to be missing error handling for the `getstreamService.getFeeds` function call. If this function throws an error, it will cause an unhandled promise rejection. It's a good practice to wrap all async operations in try-catch blocks.

```diff
module.exports = async (req, res) => {
  const {postId} = req.params;
  const token = req.token;
  const userId = req.userId;
+ 14:   let getstreamQueryResult;
+ 15:   try {
+ 16:     getstreamQueryResult = await getstreamService.getFeeds(token, 'user_anon', {
+ 17:       ids: [postId]
+ 18:     });
+ 19:   } catch (error) {
+ 20:     console.error('Error fetching feeds:', error);
+ 21:     return res.status(500).json({
+ 22:       success: false,
+ 23:       message: 'Internal server error'
+ 24:     });
+ 25:   }
  const [feed] = getstreamQueryResult?.results || {};
  console.log('userId', feed?.actor?.id, userId);
  if (feed?.actor?.id !== userId) {
    return res.status(200).json({
      success: false,
      message: 'You are not authorized to delete this post'
    });
  }

  try {
    const getstreamDeleteResult = await getstreamService.deleteFeedById({
      feedGroup: 'user_anon',
      activityId: postId,
      userId: userId,
      token
    });
    await Getstream.feed.deleteNotificationFeed(userId, postId);
    console.log('getstreamDeleteNotificationFeed');
    // console.log(getstreamDeleteNotificationFeed);
    return res.status(200).json({
      success: getstreamDeleteResult ? true : false
    });
  } catch (e) {
    console.log('error deleting anonymous post', e);
    return res.status(200).json({
      success: false,
      message: e
    });
  }
};
```

Also, it's not a good practice to expose the raw error messages to the client as it can potentially leak sensitive information. Instead, you should log the error and send a generic error message to the client.

```diff
- 43:       message: e
+ 43:       message: 'Internal server error'
```

Lastly, using `console.log` for logging is not recommended in production code. Consider using a dedicated logging library that provides more flexibility and control over your logs.


<!-- This is an auto-generated comment by OSS CodeRabbit -->